frontend/src/components/TextPopup.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  IconButton,
+  Stack,
+  SvgIcon,
+  Typography,
+} from '@mui/material';
+import ContentEditor from './ContentEditor';
+import { textSecondaryColor } from 'theme';
+import { ReactComponent as Xbtn } from 'assets/icons/x-circle.svg';
+import { ReactComponent as ArrowLogo } from 'assets/icons/chevron-down.svg';
+import { useTranslation } from 'react-i18next';
+import { StringBank } from 'strings';
+import { JSONContent } from '@tiptap/react';
+import { useState } from 'react';
+
+interface dialogProps {
+  isOpen: boolean;
+  parentSection: string;
+  newVersionName: string;
+  cancleFn: (val: boolean) => unknown;
+  completeFn: any;
+  completeBtnText: string;
+  cancleBtnText: string;
+  variabels?: any;
+}
+
+const TextPopup = (props: dialogProps) => {
```

Comment:
Let's rename this to be more indicative (e.g. NewSectionVersionDialog or something similar)

Reply:
We agreed that this element should be generic, and not specific

Reply:
changed to textEditorPopup